src/components/editor-picker.tsx

```diff
@@ -0,0 +1,57 @@
+import { SelectControl } from '@wordpress/components';
+import { useI18n } from '@wordpress/react-i18n';
+import { useEffect, useState } from 'react';
+import { SupportedEditor, supportedEditorNames } from 'src/lib/editor';
+
+interface EditorPickerProps {
+	value: SupportedEditor;
+	onChange: ( value: SupportedEditor ) => void;
+}
+
+export const EditorPicker = ( { value, onChange }: EditorPickerProps ) => {
+	const { __ } = useI18n();
+	const [ installedApps, setInstalledApps ] = useState< {
+		vscode: boolean | null;
+		phpstorm: boolean | null;
+	} >( {
+		vscode: null,
+		phpstorm: null,
+	} );
+
+	useEffect( () => {
+		const fetchInstalledApps = async () => {
+			try {
+				const apps = await window.ipcApi.getInstalledApps();
+				setInstalledApps( apps );
+			} catch ( error ) {
+				console.error( 'Failed to fetch installed apps:', error );
+			}
+		};
+
+		fetchInstalledApps();
+	}, [] );
+
+	const options = Object.entries( supportedEditorNames ).map( ( [ editor, label ] ) => {
+		const editorKey = editor as SupportedEditor;
+		const isInstalled = installedApps[ editorKey as keyof typeof installedApps ];
+
+		return {
+			value: editorKey,
+			label,
+			disabled: ! isInstalled,
+		};
+	} );
```

Comment:
Should we sort the list of installed apps by their installation status?

Reply:
Addressed in https://github.com/Automattic/studio/pull/1220